### PR TITLE
test(extension): e2e - add tests for ada handles on wallet address page

### PR DESCRIPTION
**Pull request by ljagiela**
# Allure report
`allure-report-publisher` generated test report!

**smokeTests**: ❌ [test report](https://lace-qa:8PP5wBaDV6UoXj@dq4ajm5i5q7bz.cloudfront.net/linux/chrome/2046/6198909103/index.html) for [0ea3fde2](https://github.com/input-output-hk/lace/pull/536/commits/0ea3fde2d3876438d93a3fdd11b7d6bea7bc25af)
|       | passed | failed | skipped | flaky | total | result |
|-------|--------|--------|---------|-------|-------|--------|
| Total | 30     | 2      | 0       | 0     | 32    | ❌     |

### CODE REVIEW

**Review by bslabiak — APPROVED** [review]

File: packages/e2e-tests/src/assert/walletAddressPageAssert.ts [hunk 1]
```diff
@@ -53,6 +55,22 @@ class WalletAddressPageAssert {
...
+    const handleCard = await WalletAddressPage.getHandleAddressCard(handleName);
+    await handleCard.waitForDisplayed();
+    await handleCard.$(WalletAddressPage.HANDLE_IMAGE).waitForDisplayed();
+    await handleCard.$(WalletAddressPage.HANDLE_NAME).waitForDisplayed();
+    await handleCard.$(WalletAddressPage.HANDLE_SYMBOL).waitForDisplayed();
+  }
+
+  async assertSeeTheShortestHandleFirst() {
+    const names: string[] = testContext.load('displayedAdaHandleNames');
+    await expect(names[0]).equals(Asset.ADA_HANDLE_1.name);
```
Comment (wklos-iohk) [inline packages/e2e-tests/src/assert/walletAddressPageAssert.ts:69]: redundant `await` keyword

File: packages/e2e-tests/src/data/Asset.ts [hunk 2]
```diff
@@ -11,6 +11,11 @@ export class Asset {
   public static readonly HAPPY_COIN = new Asset('HappyCoin', 'HAPP');
   public static readonly SUNDAE = new Asset('SUNDAE', 'SUNDAE');
   public static readonly HOSKY_TOKEN = new Asset('HOSKY Token', 'HOSKY');
+  public static readonly ADA_HANDLE_1 = new Asset('$cde', '$cde');
+  public static readonly ADA_HANDLE_2 = new Asset('$t_h_1', '$t_h_1');
```
Comment (wklos-iohk) [inline packages/e2e-tests/src/data/Asset.ts:15]: Most of those entries seem to be unused

File: packages/e2e-tests/src/elements/walletAddressPage.ts [hunk 3]
```diff
@@ -38,6 +43,19 @@ class WalletAddressPage extends CommonDrawerElements {
...
+  async getHandleAddressCard(handleName: string): Promise<WebdriverIO.Element> {
+    return (await this.addressCards.find(
+      async (item) => (await item.$(this.HANDLE_NAME).getText()) === handleName
+    )) as WebdriverIO.Element;
+  }
+
+  async clickCopyButtonForHandle(handleName: string) {
+    const addressCard = await this.getHandleAddressCard(handleName);
+    await addressCard.scrollIntoView();
+    await addressCard.moveTo();
```
Comment (wklos-iohk) [inline packages/e2e-tests/src/elements/walletAddressPage.ts:56]: `moveTo` does the scrolling as well if needed, see https://webdriver.io/docs/api/element/moveTo/

File: packages/e2e-tests/src/features/e2e/SendTransactionSimpleExtendedE2E.feature [hunk 4]
```diff
@@ -32,7 +32,7 @@ Feature: Send Simple Transactions - Extended view - E2E
   @LW-4677
   Scenario: Extended-view - Self Transaction E2E
     And I click "Receive" button on page header
-    And I click "Copy" button on "Wallet Address" page
+    And I click "Copy" button on "Wallet Address" page for default wallet address
```
Comment (wklos-iohk) [inline packages/e2e-tests/src/features/e2e/SendTransactionSimpleExtendedE2E.feature:35]: step not found

File: packages/e2e-tests/src/features/e2e/SendTransactionSimplePopupE2E.feature [hunk 5]
```diff
@@ -32,7 +32,7 @@ Feature: Send Simple Transactions - Popup view - E2E
   @LW-4678
   Scenario: Popup-view - Self Transaction E2E
     And I click "Receive" button on Tokens page in popup mode
-    And I click "Copy" button on "Wallet Address" page
+    And I click "Copy" button on "Wallet Address" page for default wallet address
```
Comment (wklos-iohk) [inline packages/e2e-tests/src/features/e2e/SendTransactionSimplePopupE2E.feature:35]: step not found

File: packages/e2e-tests/src/steps/waletAddressPageSteps.ts [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+  /^I see "Wallet Address" page in (extended|popup) mode for wallet "([^"]*)"$/,
+  async (mode: 'extended' | 'popup', testWalletName: string) => {
+    await walletAddressPageAssert.assertSeeWalletAddressPage(mode);
+    await walletAddressPageAssert.assertSeeWalletNameAndAddress(getTestWallet(testWalletName), mode);
+  }
+);
+
+When(/^I click "Copy" button on "Receive" page for default wallet address$/, async () => {
+  await walletAddressPage.addressCard.scrollIntoView();
+  await walletAddressPage.addressCard.moveTo();
```
Comment (wklos-iohk) [inline packages/e2e-tests/src/steps/waletAddressPageSteps.ts:23]: moveTo does the scrolling as well if needed, see https://webdriver.io/docs/api/element/moveTo/

**Review by wklos-iohk — CHANGES_REQUESTED** [review]

File: packages/e2e-tests/src/data/Asset.ts (hunk 2 above)
Comment (ljagiela) [reply packages/e2e-tests/src/data/Asset.ts:15]: I'd leave it as is for future tests and for others to know that those assets are available.

File: packages/e2e-tests/src/elements/walletAddressPage.ts (hunk 3 above)
Comment (ljagiela) [reply packages/e2e-tests/src/elements/walletAddressPage.ts:56]: It does not :) At least in practise. When I tried without scroll to WDIO was trying to moveTo to invisible element (below scroll). Tests were failing.

File: packages/e2e-tests/src/features/e2e/SendTransactionSimpleExtendedE2E.feature (hunk 4 above)
Comment (ljagiela) [reply packages/e2e-tests/src/features/e2e/SendTransactionSimpleExtendedE2E.feature]: fixed, nice catch

File: packages/e2e-tests/src/features/e2e/SendTransactionSimplePopupE2E.feature (hunk 5 above)
Comment (ljagiela) [reply packages/e2e-tests/src/features/e2e/SendTransactionSimplePopupE2E.feature]: fixed, nice catch

File: packages/e2e-tests/src/steps/waletAddressPageSteps.ts (hunk 6 above)
Comment (ljagiela) [reply packages/e2e-tests/src/steps/waletAddressPageSteps.ts:23]: see my comment above

**Review by wklos-iohk — APPROVED** [review]

**Pull request merged**